operator-framework-quarkus-extension/deployment/src/main/java/io/javaoperatorsdk/quarkus/extension/deployment/QuarkusExtensionProcessor.java

```diff
@@ -62,6 +73,78 @@ void indexSDKDependencies(
         new IndexDependencyBuildItem("io.javaoperatorsdk", "operator-framework-core"));
   }
 
+  @BuildStep
+  void createConfigurationServiceAndOperator(
+      CombinedIndexBuildItem combinedIndexBuildItem,
+      ObserverRegistrationPhaseBuildItem observerRegistrationPhase,
+      BuildProducer<ObserverConfiguratorBuildItem> observerConfigurators) {
+
+    final var index = combinedIndexBuildItem.getIndex();
+    for (ClassInfo info : index.getAllKnownImplementors(RESOURCE_CONTROLLER)) {
+      // retrieve the Controller annotation if it exists
+      final var controllerAnnotation = info.classAnnotation(CONTROLLER);
+
+      if (controllerAnnotation != null) {
+        AnnotationValue value = controllerAnnotation.value("delayRegistrationUntilEvent");
+        if (value != null) {
+          Type type = value.asClass();
+          if (type.kind() != Kind.VOID) {
+            ObserverConfigurator configurator =
+                observerRegistrationPhase
+                    .getContext()
+                    .configure()
+                    .observedType(type)
+                    .beanClass(
+                        DotName.createSimple(info.name().toString() + "_registration_observer"))
+                    .notify(
+                        mc -> {
+                          MethodDescriptor cdiMethod =
+                              MethodDescriptor.ofMethod(CDI.class, "current", CDI.class);
+                          MethodDescriptor selectMethod =
+                              MethodDescriptor.ofMethod(
+                                  CDI.class,
+                                  "select",
+                                  Instance.class,
+                                  Class.class,
+                                  Annotation[].class);
+                          MethodDescriptor getMethod =
+                              MethodDescriptor.ofMethod(Instance.class, "get", Object.class);
+                          ResultHandle cdi1 = mc.invokeStaticMethod(cdiMethod);
+                          ResultHandle operatorInstance =
+                              mc.invokeVirtualMethod(
+                                  selectMethod,
+                                  cdi1,
+                                  mc.loadClass(Operator.class),
+                                  mc.newArray(Annotation.class, 0));
+                          ResultHandle operator =
+                              mc.checkCast(
+                                  mc.invokeInterfaceMethod(getMethod, operatorInstance),
+                                  Operator.class);
+                          ResultHandle cdi2 = mc.invokeStaticMethod(cdiMethod);
```

Comment:
Any reason why `CDI.current()` is re-invoked?

Reply:
Good question.